File: src/no/omb/bigboat/data/RaceEntry.java
package no.omb.bigboat.data;

import java.text.DecimalFormat;

import no.omb.bigboat.BigBoat;

public class RaceEntry implements Comparable<RaceEntry> {

	private Boat boat;
	private int placeNo;
	private int classSize;
	private double score;

	public Boat getBoat() {
		return boat;
	}

	public void setBoat(Boat boat) {
		this.boat = boat;
	}

	public int getPlaceNo() {
		return placeNo;
	}

	public void setPlaceNo(int placeNo) {
		this.placeNo = placeNo;
	}

	public int getClassSize() {
		return classSize;
	}

	public void setClassSize(int classSize) {
		this.classSize = classSize;
	}

	public double getScore() {
		return score;
	}

	public void setScore(double score) {
		this.score = score;
	}

	public void applyValues(RaceEntry entry) {
		this.boat = entry.getBoat();
		this.classSize = entry.getClassSize();
		this.placeNo = entry.getPlaceNo();
		this.score = entry.getScore();
	}

	public String toString() {
		DecimalFormat df = new DecimalFormat("#0.00");
		return boat.toString() + BigBoat.SEP + placeNo + BigBoat.SEP + df.format(score);
	}

	@Override
	public int compareTo(RaceEntry o) {
		return (int) (score*100 - o.score*100);
	}
}

File: src/main/kotlin/no/omb/bigboat/data/RaceEntry.kt
package no.omb.bigboat.data

import no.omb.bigboat.BigBoat
import java.text.DecimalFormat

data class RaceEntry(
    var boat: Boat? = null,
    var placeNo: Int = 0,
    var classSize: Int = 0,
    var score: Double = 0.0
) : Comparable<RaceEntry> {
    fun applyValues(entry: RaceEntry) {
        boat = entry.boat
        classSize = entry.classSize
        placeNo = entry.placeNo
        score = entry.score
    }

    override fun toString(): String {
        val df = DecimalFormat("#0.00")
        return boat.toString() + BigBoat.SEP + placeNo + BigBoat.SEP + df.format(score)
    }

    override fun compareTo(other: RaceEntry): Int {
        return (score * 100 - other.score * 100).toInt()
    }
}
